gemhardware/devices/src/common/amc/HwGenericAMC.cc

```diff
@@ -608,343 +470,257 @@ void gem::hw::HwGenericAMC::setDAQLinkRunParameter(uint8_t const& parameter, uin
-            XCEPT_RAISE(gem::hw::devices::exception::RPCMethodError, errmsg.str());
-          }
-        } STANDARD_CATCH;
-        return rsp.get_word("phase");
-      } STANDARD_CATCH;
-    } GEM_CATCH_RPC_ERROR("HwGenericAMC::getGTHPhaseMean", gem::hw::devices::exception::Exception);
-  }
+  float res = 0.;
+  try {
+    res = xhal::common::rpc::call<::amc::ttc::getGTHPhaseMean>(rpc, readAttempts);
+  } GEM_CATCH_RPC_ERROR("HwGenericAMC::getGTHPhaseMean", gem::hw::devices::exception::Exception);
+  return static_cast<double>(res);
 }
 
-double gem::hw::HwGenericAMC::getGTHPhaseMedian(uint32_t readAttempts)
+double gem::hw::amc::HwGenericAMC::getGTHPhaseMedian(uint32_t readAttempts)
 {
   CMSGEMOS_WARN("HwGenericAMC::getGTHPhaseMedian is not implemented, returning the mean");
   return getGTHPhaseMean(readAttempts);
 }
 
-void gem::hw::HwGenericAMC::ttcCounterReset()
+void gem::hw::amc::HwGenericAMC::ttcCounterReset()
 {
-  writeReg(getDeviceBaseNode(), "TTC.CTRL.CNT_RESET", 0x1);
+  writeReg("GEM_AMC.TTC.CTRL.CNT_RESET", 0x1);
 }
 
-bool gem::hw::HwGenericAMC::getL1AEnable()
+bool gem::hw::amc::HwGenericAMC::getL1AEnable()
 {
-  return readReg(getDeviceBaseNode(), "TTC.CTRL.L1A_ENABLE");
+  return readReg("GEM_AMC.TTC.CTRL.L1A_ENABLE");
 }
 
-void gem::hw::HwGenericAMC::setL1AEnable(bool enable)
+void gem::hw::amc::HwGenericAMC::setL1AEnable(bool enable)
 {
   // uint32_t safeEnable = 0xa4a2c200+int(enable);
-  writeReg(getDeviceBaseNode(), "TTC.CTRL.L1A_ENABLE", uint32_t(enable));
+  writeReg("GEM_AMC.TTC.CTRL.L1A_ENABLE", uint32_t(enable));
 }
 
-uint32_t gem::hw::HwGenericAMC::getTTCConfig(AMCTTCCommandT const& cmd)
+uint32_t gem::hw::amc::HwGenericAMC::getTTCConfig(AMCTTCCommandT const& cmd)
 {
   CMSGEMOS_WARN("HwGenericAMC::getTTCConfig: not implemented");
   return 0x0;
 }
 
-void gem::hw::HwGenericAMC::setTTCConfig(AMCTTCCommandT const& cmd, uint8_t const& value)
+void gem::hw::amc::HwGenericAMC::setTTCConfig(AMCTTCCommandT const& cmd, uint8_t const& value)
 {
   CMSGEMOS_WARN("HwGenericAMC::setTTCConfig: not implemented");
   return;
 }
 
-uint32_t gem::hw::HwGenericAMC::getTTCStatus()
+uint32_t gem::hw::amc::HwGenericAMC::getTTCStatus()
 {
   CMSGEMOS_WARN("HwGenericAMC::getTTCStatus: not fully implemented");
-  return readReg(getDeviceBaseNode(), "TTC.STATUS");
+  return readReg("GEM_AMC.TTC.STATUS");
 }
 
-uint32_t gem::hw::HwGenericAMC::getTTCErrorCount(bool const& single)
+uint32_t gem::hw::amc::HwGenericAMC::getTTCErrorCount(bool const& single)
 {
   if (single)
-    return readReg(getDeviceBaseNode(), "TTC.STATUS.TTC_SINGLE_ERROR_CNT");
+    return readReg("GEM_AMC.TTC.STATUS.TTC_SINGLE_ERROR_CNT");
   else
-    return readReg(getDeviceBaseNode(), "TTC.STATUS.TTC_DOUBLE_ERROR_CNT");
+    return readReg("GEM_AMC.TTC.STATUS.TTC_DOUBLE_ERROR_CNT");
 }
 
-uint32_t gem::hw::HwGenericAMC::getTTCCounter(AMCTTCCommandT const& cmd)
+uint32_t gem::hw::amc::HwGenericAMC::getTTCCounter(AMCTTCCommandT const& cmd)
```

Comment:
but in cases like this, where there is much more than just a single `(read|write)Reg` it would have made sense to use the `ctp7_modules` functor call

Reply:
yes, but this is out of the scope of the current PR